### ISSUE TITLE
relative path cause `invalid path` error when used in build in browser

### ISSUE THREAD

**Issue by hardfist** [evanw/esbuild#693]
when I use esbuild in browser, if outfile  is relative path （even though write is false) will cause error, which seems caused by golang doesn't implement fs.stat when target GOOS=js
```
async function main() {
  const service = await esbuild.startService({
    wasmURL: 'https://unpkg.com/esbuild-wasm@0.8.33/esbuild.wasm',
  });
  const result = await service.build({
    stdin: {
      contents: `const x = 1`,
    },
    write: false,
    outfile: 'esbuild.js',
  });
  console.log('result:', result);
}

main();

```
![image](https://user-images.githubusercontent.com/8898718/104989101-a95b0780-5a54-11eb-9d2d-a444295f8c84.png)

**Comment by evanw** [comment]
Thanks for the heads up. For now, I believe a workaround is to use an absolute path. I need to figure out what to do about paths with WebAssembly. It turns out [Go's path library is pretty broken when targeting WebAssembly](https://github.com/evanw/esbuild/issues/687) so I may need to fork the standard library and implement path manipulation myself. Fixing that would make it easy to fix this too.

**Comment by evanw** [comment]
This should be fixed as of version 0.8.35.
